frontend/test/metabase/scenarios/dashboard/dashboard-drill.cy.spec.js

```diff
@@ -171,75 +171,92 @@ describe("scenarios > dashboard > dashboard drill", () => {
+        database: 1,
+        query: {
+          "source-table": 4,
+        },
+        type: "query",
+      },
+      display: "table",
+      visualization_settings: {},
+    }).then(({ body: { id: questionId } }) => {
+      // 3. create a dashboard
+      cy.request("POST", "/api/dashboard", {
+        name: "13062D",
+      }).then(({ body: { id: dashboardId } }) => {
+        // add filter to the dashboard
+        cy.request("PUT", `/api/dashboard/${dashboardId}`, {
+          parameters: [
+            {
+              id: "18024e69",
+              name: "Category",
+              slug: "category",
+              type: "category",
+            },
+          ],
+        });
 
-    // add values to the filter
-    cy.findByText("Category").click();
-    popover().within(() => {
-      cy.findByText("5").click();
-      cy.findByText("4").click();
-    });
-    cy.findByText("Add filter").click();
+        // add previously created question to the dashboard
+        cy.request("POST", `/api/dashboard/${dashboardId}/cards`, {
+          cardId: questionId,
+        }).then(({ body: { id: dashCardId } }) => {
+          // connect filter to that question
+          cy.request("PUT", `/api/dashboard/${dashboardId}/cards`, {
+            cards: [
+              {
+                id: dashCardId,
+                card_id: questionId,
+                row: 0,
+                col: 0,
+                sizeX: 8,
+                sizeY: 6,
+                parameter_mappings: [
+                  {
+                    parameter_id: "18024e69",
+                    card_id: questionId,
+                    target: ["dimension", ["field-id", 31]], // 31 = REVIEWS.RATING
+                  },
+                ],
+              },
+            ],
+          });
+        });
 
-    // drill-through
-    cy.findByText("xavier").click();
-    cy.findByText("=").click();
+        // NOTE: The actual "Assertion" phase begins here
+        cy.log("**Reported failing on Metabase 1.34.3 and 0.36.2**");
 
-    cy.log("**Reported failing on Metabase 1.34.3 and 0.36.2**");
-    cy.findByText("Reviewer is xavier");
-    cy.findByText("Rating is equal to 2 selections");
-    // wait for data to finish loading
-    cy.get(".LoadingSpinner").should("not.exist");
+        cy.log("**The first case**");
+        // set filter values (ratings 5 and 4) directly through the URL
+        cy.visit(`/dashboard/${dashboardId}?category=5&category=4`);
 
-    cy.log("**Test the second case reported in this issue**");
-    // go back to the dashboard
-    cy.visit("/dashboard/6?category=5&category=4");
-    cy.findByText("2 selections");
+        // drill-through
+        cy.findByText("xavier").click();
+        cy.findByText("=").click();
 
-    cy.findByText("Reviews").click(); // the card title
-    cy.findByText("Rating is equal to 2 selections");
+        cy.findByText("Reviewer is xavier");
+        cy.findByText("Rating is equal to 2 selections");
```

Comment:
Let's also make sure the expected results show up in the page. (I don't have a great way to do this for the second test below).

```suggestion
        cy.findByText("Rating is equal to 2 selections");
        cy.contains("Reprehenderit non error");
```

Reply:
We can use the same `cy.contains("Reprehenderit non error");` for the second case. I am fine with this - just hope there will not be any weirdness in CI. `cy.contains()` should search for that string for about 4s. I'll let this CI run finish first and then commit your suggestions to let it run again.

When it comes to flake fixes, I'd rather see 2-3 green CI runs in a row before merging.